### PR TITLE
Mobify bloodlings

### PR DESCRIPTION
**Pull request by FlameArrow57**
[AI][INTERNAL][REWORK]

## About the PR 
This PR makes bloodlings mob critters.

## Why's this needed? 
Object critter removal

### CODE REVIEW

**Comment by CheffieGithub** [comment]
You have deleted the ketchupling fully, please convert that too

File: code/mob/living/critter/bloodling.dm [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		var/datum/healthHolder/brute = src.add_health_holder(/datum/healthHolder/bloodling_brute)
+		brute.value = 25
+		brute.maximum_value = 25
+		brute.last_value = 25
+		brute.damage_multiplier = 0.5
+
+		var/datum/healthHolder/burn = src.add_health_holder(/datum/healthHolder/bloodling_burn)
+		burn.value = 25
+		burn.maximum_value = 25
+		burn.last_value = 25
```
Comment (CheffieGithub) [inline code/mob/living/critter/bloodling.dm:58]: any reason not to use add_hh_flesh and add_hh_flesh_burn here?

**Comment by FlameArrow57** [comment]
Wasn't aware of any cases where it was used

File: code/mob/living/critter/bloodling.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/mob/living/critter/bloodling.dm:58]: It has stuff about chems which I don't think bloodling would use really

**Comment by CheffieGithub** [comment]
its currently used for basically all critters use add_hh_brute(amount, vuln) and likewise for burn

**Comment by FlameArrow57** [comment]
Oops, was meaning ketchupling

**Comment by CheffieGithub** [comment]
admin gimmick thing that pali made, still convert theres no reason not to

File: code/mob/living/critter/bloodling.dm (hunk 1 above)
Comment (CheffieGithub) [reply code/mob/living/critter/bloodling.dm:58]: they are still used for critters that don't metabolise

**Review by CheffieGithub — APPROVED** [review]

File: code/mob/living/critter/ai/bloodling.dm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/attack/bloodling, list(src.holder, src))
+
+/datum/aiTask/timed/wander/critter/aggressive/bloodling
+
+/datum/aiTask/timed/wander/critter/aggressive/bloodling/on_tick()
+	if (prob(10))
+		var/mob/living/critter/bloodling/bloodling = src.holder.owner
+		if (!(locate(bloodling.cleanable_type) in src.holder.owner.loc))
+			playsound(src.holder.owner.loc, 'sound/impact_sounds/Slimy_Splat_1.ogg', 30, TRUE, -1)
+			new bloodling.cleanable_type(get_turf(src.holder.owner))
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/ai/bloodling.dm:18]: Seems odd to do this from the AI code, why not do it in `Life` on the mob so it works when a player is controlling it too?

File: code/mob/living/critter/ai/bloodling.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	if (narrator_mode)
+		playsound(bloodling.loc, 'sound/vox/ghost.ogg', 50, TRUE, -1)
+	else
+		playsound(bloodling.loc, 'sound/effects/ghost.ogg', 30, TRUE, -1)
+	if (prob(50))
+		var/mob/living/carbon/C = src.holder.target
+		boutput(C, "<span class='combat'><b>You are forced to the ground by \the [bloodling]!</b></span>")
+		random_brute_damage(C, rand(0, 5))
+		C.changeStatus("stunned", 5 SECONDS)
+		C.changeStatus("weakened", 5 SECONDS)
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/ai/bloodling.dm:44]: `failed` is supposed to return `TRUE` if the succeedable task has failed in some way, not handle attack behaviour afaik

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Comment by FlameArrow57** [comment]
Probably not going to be finishing this soon so just closing, anyone can take code tho

**Pull request closed without merging**